packages/ckeditor5-utils/src/version.ts

```diff
@@ -24,138 +24,162 @@ declare global {
+	 *
+	 * These are the most common import methods of the CKEditor&nbsp;5 packages.
+	 *
+	 * - **NIM (New installation methods)** &ndash; Imports from the `ckeditor5` and `ckeditor5-premium-features` packages.
+	 * - **Optimized build** for the new installation methods &ndash; Imports from the `@ckeditor/ckeditor5-<NAME>/dist/index.js`.
+	 * - **Predefined builds** (legacy) &ndash; Imports from the `@ckeditor/ckeditor5-build-<NAME>` packages.
+	 * - **Default imports** (legacy) &ndash; Imports from the `@ckeditor/ckeditor5-<NAME>` packages (default export).
+	 * - **`src`** (legacy) &ndash; Imports from the `@ckeditor/ckeditor5-<NAME>/src/*`.
+	 * - **DLL builds** (legacy) &ndash; Imports from the `ckeditor5/build/<NAME>` and `@ckeditor/ckeditor5-<NAME>/build/*`.
+	 *
+	 * The best way to avoid duplicate modules is to not mix these installation methods. For example, if you use imports
+	 * specific to the optimized build, you should use them for all CKEditor&nbsp;5 packages. In addition, since
+	 * the Predefined and DLL builds already include the core of the editor, they cannot be used with other types of imports.
+	 *
+	 * This is a matrix showing which installation methods are compatible with each other:
+	 *
+	 * |                  | NIM | Optimized build | Predefined builds | Default imports | `src` | DLL builds |
+	 * |------------------|-----|-----------------|-------------------|-----------------|-------|------------|
+	 * | NIM              | ✅  | ✅              | ❌                | ❌              | ❌    | ❌         |
+	 * | Optimized builds | ✅  | ✅              | ❌                | ❌              | ❌    | ❌         |
+	 * | Predefined build | ❌  | ❌              | ✅                | ❌              | ❌    | ❌         |
+	 * | Default imports  | ❌  | ❌              | ❌                | ✅              | ✅    | ❌         |
+	 * | `src`            | ❌  | ❌              | ❌                | ✅              | ✅    | ❌         |
+	 * | DLL builds       | ❌  | ❌              | ❌                | ❌              | ❌    | ✅         |
+	 *
+	 * If you use any third-party plugins, then make sure the way you import them is compatible with
+	 * the way you import CKEditor&nbsp;5.
+	 *
+	 * <details>
+	 * <summary>New installation methods and optimized builds</summary>
+	 *
+	 * If you are using the {@glink updating/nim-migration/migration-to-new-installation-methods new installation methods},
+	 * you should only import code from the `ckeditor5` and `ckeditor5-premium-features` packages.
+	 * Do not import code from the `@ckeditor/ckeditor5-<NAME>` packages unless you follow
+	 * the {@glink getting-started/setup/optimizing-build-size Optimizing build size} guide and the imports from
+	 * the `@ckeditor/ckeditor5-<NAME>` packages end with `/dist/index.js`.
+	 *
+	 * If you are using a CDN, make sure that some files are not included twice in your project.
+	 *
+	 * Examples:
+	 *
```

Comment:
This kinda looks as if importing Highlights twice was an OK action to do.

Reply:
I changed `Examples` to `Examples of valid and invalid import paths` to better explain this. Do you think it's enough?

Reply:
Yeah, it's explicit now.